### PR TITLE
BEVFormerV2 Functional

### PR DESCRIPTION
**Pull request by ign-saurav**
## What's changed

Added Tenstorrent Neural Network implementation of **BEVFormerV2** in `tt-metal` with a full multi-view 3D detection pipeline (backbone, FPN, perception transformer, and BEV head) running on **Wormhole (n150)**.

- **Input Resolution:** 256×704 – multi-view images from 6 cameras (front, front-left, front-right, back, back-left, back-right)  
- **BEV Grid:** 200×200 
- **Model:** BEVFormerV2 (900 queries, 10 nuScenes classes, 256-dim features)  
- **Reference:** [BEVFormer](https://github.com/fundamentalvision/BEVFormer) 

---

### PCC (TTNN vs PyTorch)

| Output          | PCC   |
|-----------------|-------|
| bev_embed       | 0.99 |
| all_cls_scores  | 0.99  |
| all_bbox_preds  | 0.99  |

- PCC checks use a threshold of **0.97**.  

### Performance (Wormhole n150, BS=1)

- **Device FPS** 0.102 
- **E2E FPS** 0.08

### Checklist

- [ ] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml?query=branch:ign/BEVFormerV2)
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:ign/BEVFormerV2)
- [ ] [cpp-unit-tests](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ign/BEVFormerV2)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/experimental/BEVFormerV2/README.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+### Run Device Performance Test
+
+```bash
+pytest models/experimental/BEVFormerV2/tests/perf/test_device_perf.py -s
+```
+
+### Run End-to-End Performance Test
+
+```bash
+pytest models/experimental/BEVFormerV2/tests/perf/test_e2e_performant_2cq_no_trace.py -s
```
Comment (dvartaniansTT) [inline models/experimental/BEVFormerV2/README.md:155]: why no trace?

File: models/experimental/BEVFormerV2/README.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+- **Checkpoint Location:** Auto-downloaded via `common.py` on first use
+
+Note: The weights are trained on the nuScenes dataset.
+
+## Quickstart
+
+### Run Tests
+
+```bash
+pytest models/experimental/BEVFormerV2/tests/pcc/test_bevformer_v2.py
```
Comment (dvartaniansTT) [inline models/experimental/BEVFormerV2/README.md:93]: I had to do: 
`pip install pyquaternion --target /home/ubuntu/ign/tt-metal/python_env/lib/python3.10/site-packages`

add this to README instructions.

File: models/experimental/BEVFormerV2/README.md (hunk 1 above)
Comment (dvartaniansTT) [inline models/experimental/BEVFormerV2/README.md:155]: the perf test failed for me: 

Image

**Review by dvartaniansTT — CHANGES_REQUESTED** [review]
1. fix the failing perf test
2. why there is no trace implementation?
3. add missing pip install instructions to the README. 

Good work so far and good visual demo
